### PR TITLE
Trying to implement nested route in App.js

### PR DESCRIPTION
**Pull request by jfan0010**
# Description 
I've tried to implement routes with different layouts in App.js, which can let us use different layout, yet keeping the App.js simple.
1.  Refactor the file of `component/layout` to `component/layouts/...`, which inside there're 
1.1 `.../layoutComponents`  - where we store header/sidebar and other kind of header/sidebar that we are adding in future
1.2 `.../dashboardLayout`  - where we can get the normal researchify dashboard layout, with header and sidebar
1.3 `.../editorLayout`  - where we can get the editor layout without header
1.4 if we're adding more layout in future for client site and reasearchify site we can add those layouts in this folder
2. In the layout files, I've created a `DashboardLayoutRoute` and `EditorLayoutRoute` which can be used and replace the normal `<Route>` in App.js (passing the same props into it).

# Type of change 
- [ ] Bug fix
- [ ] New feature
- [x] Refactoring
- [ ] Documentation 
- [x] Spikes. To compare the different of current branch.

# Problem
App.js is messy when using different layout

# Solution description
Implemented routes with different layouts, each layout will have their own routes tag

# Tests 
All unit tests pass?

Builds successfully?

Attach screenshot of the expected result (e.g. UI of the website/ postman api call result/ database changes)

Provide steps in detail for reviewers to reproduce the expected result of the code changes.

# Relevant document/ link (if any) 
Check this link for basic nested routes: [Official React-Router Site](https://reactrouter.com/web/guides/quick-start#:~:text=2nd,value)  
Example in this article is adapted into the code: [Article from Selva Kumar](https://dev.to/selvaece25/multiple-layout-in-react-router-3acb)  
The GitHub Repo of final code of Selva Kumar: [GitHub of The Entire Demo](https://github.com/selvaece25/react-all-in-one)

# Risk
Identity the risk level (low, medium, high) and indicate which component(s) will be affected if bugs exist.

### CODE REVIEW

File: client/src/components/App.js [hunk 1]
```diff
@@ -4,27 +4,23 @@
...
-import Login from './auth/Login';
-import Layout from './layout/Layout';
-import SidebarData from './layout/SidebarData'
+import './layouts/Layout.css';
 
-import './layout/Layout.css'
-import PublicationPage from './publications/PublicationPage';
+/** Layout **/
+import DashboardLayoutRoute from './layouts/dashboardLayout/DashboardLayoutRoute';
+import EditorLayoutRoute from './layouts/editorLayout/EditorLayoutRoute';
```
Comment (jfan0010) [inline client/src/components/App.js:21]: arranged the imports

File: client/src/components/App.js [hunk 2]
```diff
@@ -4,27 +4,23 @@
...
 
 const App = () => {
-    
-    const headerData = {
-        title: "Researchify",
-        dashboardURL: "/dashboard",
-        profileURL: "/dashboard/profile"
-    }
```
Comment (jfan0010) [inline client/src/components/App.js:27]: headerData putted in DashboardLayout.js

File: client/src/components/App.js [hunk 3]
```diff
@@ -33,15 +29,14 @@ const App = () => {
...
-                    <Route path="/dashboard/profile" exact component={ProfileInfoEdit} />
-                </Layout>
-
-            </Switch>             
+
+                <DashboardLayoutRoute path="/dashboard" exact component={Dashboard} />
+                <DashboardLayoutRoute path="/dashboard/profile" exact component={ProfileInfoEdit} />
+                <DashboardLayoutRoute path="/publications/team" exact component={PublicationPage} />
+
+                <EditorLayoutRoute path="/editor" exact component={EditorHome} />
```
Comment (jfan0010) [inline client/src/components/App.js:37]: DashboardLayoutRoute is the route with dashboard layout, EditorLayoutRoute is with EditorLayout

File: client/src/components/App.js (hunk 1 above)
Comment (rpra0006) [reply client/src/components/App.js:21]: I like this arrangement

File: client/src/components/dashboard.js [hunk 4]
```diff
@@ -1,12 +0,0 @@
...
- */
-import React from 'react';
-import {useSelector} from 'react-redux';
-
-export default function Dashboard() {
-    const userName = useSelector(state => {console.log(state); return state.user?.givenName});
-    return (
-        <div>Hi {userName}!</div>
-    )
-}
```
Comment (rpra0006) [inline client/src/components/dashboard.js:12]: Is this dashboard.js team Logiciel's commit? Are we deleting it, if we are you can ignore this comment.

File: client/src/components/layouts/dashboardLayout/DashboardLayout.js [hunk 5]
```diff
@@ -0,0 +1,33 @@
+import React, { Fragment } from 'react';
+import Sidebar from '../layoutComponents/Sidebar';
+import Header from '../layoutComponents/Header';
+import { Container, Col, Row } from 'react-bootstrap';
+import SidebarData from '../SidebarData';
```
Comment (rpra0006) [inline client/src/components/layouts/dashboardLayout/DashboardLayout.js:5]: We can probably rename this to DashboardSidebarData to keep consistency with EditorSidebarData naming convention.

File: client/src/components/layouts/dashboardLayout/DashboardLayoutRoute.js [hunk 6]
```diff
@@ -0,0 +1,20 @@
+import React from 'react';
+import { Route } from 'react-router-dom';
+import DashboardLayout from './DashboardLayout';
+
+const DashboardLayoutRoute = ({ component: Component, ...rest}) => {
+    //get 'component' and renamed to 'Component', any other 'props' renamed to 'rest'
+    return (
+        <Route
```
Comment (rpra0006) [inline client/src/components/layouts/dashboardLayout/DashboardLayoutRoute.js:8]: Is this part what you meant by nested route?

File: client/src/components/layouts/dashboardLayout/DashboardLayoutRoute.js [hunk 7]
```diff
@@ -0,0 +1,20 @@
+import React from 'react';
+import { Route } from 'react-router-dom';
+import DashboardLayout from './DashboardLayout';
+
+const DashboardLayoutRoute = ({ component: Component, ...rest}) => {
+    //get 'component' and renamed to 'Component', any other 'props' renamed to 'rest'
+    return (
+        <Route 
+            {...rest}
```
Comment (rpra0006) [inline client/src/components/layouts/dashboardLayout/DashboardLayoutRoute.js:9]: What does {...rest} do here?

**Review by rpra0006 — APPROVED** [review]
I got some questions inside my code review. Can you have a look at those? Thanks!

File: client/src/components/dashboard.js (hunk 4 above)
Comment (jfan0010) [reply client/src/components/dashboard.js:12]: Erm I dont think it's our commit, but i guess we took over the dashboard task after other team created a welcoming-user-dashboardpage?

File: client/src/components/layouts/dashboardLayout/DashboardLayout.js (hunk 5 above)
Comment (jfan0010) [reply client/src/components/layouts/dashboardLayout/DashboardLayout.js:5]: Make sense

File: client/src/components/layouts/dashboardLayout/DashboardLayoutRoute.js (hunk 6 above)
Comment (jfan0010) [reply client/src/components/layouts/dashboardLayout/DashboardLayoutRoute.js:8]: yes

File: client/src/components/layouts/dashboardLayout/DashboardLayoutRoute.js (hunk 7 above)
Comment (jfan0010) [reply client/src/components/layouts/dashboardLayout/DashboardLayoutRoute.js:9]: My guessing/understanding is that it only uses the `Component` which is the page rendered in the layout, and pass other `props` into `<Route>`  tag. For example when we coded `<DashboardLayoutRoute path="/publications/team" exact component={PublicationPage} />`, `props.component` will go to the `<Component>`, and `props.path` will become part of the `{...rest}` and passed into `<Route>`

File: client/src/components/layouts/dashboardLayout/DashboardLayoutRoute.js (hunk 6 above)
Comment (jfan0010) [reply client/src/components/layouts/dashboardLayout/DashboardLayoutRoute.js:8]: Well I think its not really nested route, just routes with different layouts

File: client/src/components/dashboard.js (hunk 4 above)
Comment (rpra0006) [reply client/src/components/dashboard.js:12]: Sure then if it's not used we can delete it

File: client/src/components/layouts/dashboardLayout/DashboardLayoutRoute.js (hunk 7 above)
Comment (rpra0006) [reply client/src/components/layouts/dashboardLayout/DashboardLayoutRoute.js:9]: Ok makes sense

**Pull request merged**